pkg/reconciler/pipelinerun/resources/pipelinerunstate.go

```diff
@@ -159,8 +156,9 @@ func (state PipelineRunState) GetTaskRunsResults() map[string][]v1beta1.TaskRunR
 		if !rpt.isSuccessful() {
 			continue
 		}
-		if rpt.TaskRun != nil {
-			results[rpt.PipelineTask.Name] = rpt.TaskRun.Status.TaskRunResults
+		// Currently a Matrix cannot produce results so this is for a singular TaskRun
```

Comment:
Is this true? I thought matrixed taskruns can produce results, just not consume full array results. It seems like this function is just for collecting results produced by each pipeline task, so I'm curious why it doesn't support matrix.

Reply:
Currently producing results by a matrix is not supported. Working on a design doc to implement supporting producing results. 
https://github.com/tektoncd/pipeline/blob/42d1901a46bb3e50279930c4a4ecfb96d5256b27/docs/matrix.md?plain=1#L298-L300